cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -69,6 +79,34 @@ using v1::by_default;
 
 } // namespace method
 
+namespace optional_results {
+
+class namespace_id {};
+using optional_result_id_t = optional_result_id<namespace_id>;
```

Comment:
In the email thread we agreed to remove namespace id and use inheritance

Reply:
Fixed